packages/opencensus-core/src/stats/recorder.ts

```diff
@@ -14,63 +14,62 @@
  * limitations under the License.
  */
 
-import {AggregationData, AggregationType, CountData, DistributionData, LastValueData, Measurement, SumData} from './types';
+import {AggregationData, AggregationType, CountData, DistributionData, LastValueData, Measurement, MeasureType, SumData} from './types';
 
 export class Recorder {
   static addMeasurement(
       aggregationData: AggregationData,
       measurement: Measurement): AggregationData {
     switch (aggregationData.type) {
       case AggregationType.DISTRIBUTION:
-        return this.addToDistribution(
-            aggregationData as DistributionData, measurement);
+        return this.addToDistribution(aggregationData, measurement);
 
       case AggregationType.SUM:
-        return this.addToSum(aggregationData as SumData, measurement);
+        return this.addToSum(aggregationData, measurement);
 
       case AggregationType.COUNT:
-        return this.addToCount(aggregationData as CountData, measurement);
+        return this.addToCount(aggregationData, measurement);
 
       default:
-        return this.addToLastValue(
-            aggregationData as LastValueData, measurement);
+        return this.addToLastValue(aggregationData, measurement);
     }
   }
 
   private static addToDistribution(
       distributionData: DistributionData,
       measurement: Measurement): DistributionData {
     distributionData.timestamp = Date.now();
+    const value = measurement.measure.type === MeasureType.DOUBLE ?
```

Comment:
Suggestion: have `addTo*` take in `value` directly as an argument, and do this boilerplate in `addMeasurement`.

Reply:
Ways simpler. Thanks for the suggestion.